### PR TITLE
Only see ancestors

### PR DESCRIPTION
**Pull request by emord**
@esoergel I think this covers the rest of the location functionality. I'll probably take another pass of testing though this tomorrow. 

enchantress: @nickpell

### CODE REVIEW

File: corehq/apps/locations/resources/v0_1.py [hunk 1]
```diff
@@ -20,14 +23,30 @@ def obj_get(self, bundle, **kwargs):
 
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
+        project = bundle.request.project
         parent_id = bundle.request.GET.get('parent_id', None)
         include_inactive = json.loads(bundle.request.GET.get('include_inactive', 'false'))
-        if parent_id:
+        user = bundle.request.couch_user
+
```
Comment (dimagimon) [inline corehq/apps/locations/resources/v0_1.py]: W293 blank line contains whitespace

File: corehq/apps/locations/views.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 from .permissions import (locations_access_required, is_locations_admin,
                           can_edit_location, can_edit_location_types)
-from .models import Location, LocationType, SQLLocation
+from .models import Location, LocationType, SQLLocation, all_locations
```
Comment (dimagimon) [inline corehq/apps/locations/views.py]: F401 'all_locations' imported but unused

File: corehq/apps/locations/views.py [hunk 3]
```diff
@@ -77,13 +77,16 @@ def show_inactive(self):
...
         return {
             'selected_id': selected_id,
             'locations': load_locs_json(
-                self.domain, selected_id, self.show_inactive
+                self.domain, selected_id, self.show_inactive, self.request.couch_user
             ),
             'show_inactive': self.show_inactive,
-            'has_location_types': has_location_types
+            'has_location_types': has_location_types,
+            'can_edit_root': (not loc_restricted or
```
Comment (dimagimon) [inline corehq/apps/locations/views.py]: W291 trailing whitespace

File: corehq/apps/locations/permissions.py [hunk 4]
```diff
@@ -20,11 +20,36 @@ def is_locations_admin(view_fn):
     return locations_access_required(domain_admin_required(view_fn))
 
 
-def user_can_edit_location(user, location):
-    user_loc = user.get_location(location.domain).sql_location
+def user_can_edit_location(user, location, project):
+    """
+    Expects SQLLocation
```
Comment (czue) [inline corehq/apps/locations/permissions.py]: in this instance I think rather than add docs it'd be better to rename the variable to `sql_location`

**Comment by esoergel** [comment]
Looks good overall, particularly the changes to the location tree js.  I didn't realize we were also hiding locations outside your branch though (instead of just making them not editable), are you sure that's supposed to happen?  Also, `user_can_edit_location` and `user_can_view_location` might cause some performance issues when applied en masse to the big list of locations, I imagine we'll need to come up with a more performant way to get editable and/or viewable locations in bulk, but this is probably fine for now, given the time pressure.

**Comment by emord** [comment]
@esoergel Hmm perhaps I misread this:

> When assigned to a location, a web user can only see but not edit all ancestors (location tree and location edit pages)

Seems to be an OK behavior though right? Let me know if I should change

**Comment by esoergel** [comment]
Just offlined with Sheel, let's keep it the way you've done it here.

**Pull request merged**